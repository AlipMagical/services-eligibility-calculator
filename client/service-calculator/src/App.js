/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import React, { createContext, useState } from 'react';

import { ANALYTICS_ID } from './common/config';
import { Body } from './components/Body';
import { DataContext } from './DataContext';
import Footer from './components/Footer';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import { Router } from 'react-router-dom';
import ScrollToTop from './common/utils/ScrollToTop';
import { createBrowserHistory } from 'history';

export const FilterContext = createContext();

export const createInitialFilterState = () => ({
  income: undefined,
  interests: {},
  people: {}
});

const history = createBrowserHistory();

export default function App(props) {
  const [analyticsInitialized, setAnalyticsInitialized] = useState(false);
  const [lastSearch, setLastSearch] = useState('');
  const [filterState, setFilterState] = useState(createInitialFilterState());
  const resetFilterState = () => setFilterState(createInitialFilterState());

  if (!analyticsInitialized) {
    ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });
    setAnalyticsInitialized(true);
  }

  return (
    <DataContext.StateProvider>
      <FilterContext.Provider
        value={{
          filterState,
          setFilterState,
          resetFilterState,
          lastSearch,
          setLastSearch
        }}
      >
        <Router history={history}>
          <ScrollToTop />
          <Body />
          <Footer />
        </Router>
      </FilterContext.Provider>
    </DataContext.StateProvider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
