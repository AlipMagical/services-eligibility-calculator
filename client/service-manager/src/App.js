/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import React, { createContext, useState } from 'react';
import {
  getLoggedInUser,
  getMillisUntilExpiration,
  isLoggedIn
} from './common/auth';

import { ANALYTICS_ID } from './common/config';
import { BASEPATH } from './common/config';
import Header from './components/Header';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import Toast from './components/Toast';
import departmentRoutes from './routes/department';
import getUserConfirmation from './getUserConfirmation';
import interestRoutes from './routes/interest';
import serviceRoutes from './routes/service';
import settingsRoutes from './routes/settings';
import { useSettings } from './hooks';

export const AuthContext = createContext();
export const ToastContext = createContext();
export const SettingsContext = createContext();

export default function App(props) {
  ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });

  const [loginState, setLoginState] = useState({
    isLoggedIn: isLoggedIn(),
    loggedInUser: getLoggedInUser()
  });

  const onLoginStatusChanged = () => {
    //setTimeout to update when token might expire.
    const millis = getMillisUntilExpiration();
    millis && setTimeout(onLoginStatusChanged, millis);

    setLoginState({
      isLoggedIn: isLoggedIn(),
      loggedInUser: getLoggedInUser()
    });
  };

  const { settings, setSettings } = useSettings();

  const [toastMessage, setToastMessage] = useState(null);

  return (
    <AuthContext.Provider value={{ ...loginState, onLoginStatusChanged }}>
      <SettingsContext.Provider value={{ settings, setSettings }}>
        <ToastContext.Provider value={{ toastMessage, setToastMessage }}>
          <BrowserRouter getUserConfirmation={getUserConfirmation}>
            <Header />
            <div className="container bd-content">
              <Switch>
                {serviceRoutes}
                {settingsRoutes}
                {interestRoutes}
                {departmentRoutes}
                <Redirect from={BASEPATH} exact to={BASEPATH + 'services/'} />
                <Route render={() => <h1>Not Found</h1>} />
              </Switch>
            </div>
            {toastMessage && <Toast message={toastMessage} />}
          </BrowserRouter>
        </ToastContext.Provider>
      </SettingsContext.Provider>
    </AuthContext.Provider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
