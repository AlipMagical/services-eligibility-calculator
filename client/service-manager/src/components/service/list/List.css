/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.search-match {
  font-weight: bold;
}

.service-draft-name {
  font-size: 18px;
}

.draft-shared-by {
  margin-bottom: 0px;
}

.draft-shared-by > i {
  margin-right: 10px;
}

.draft-last-updated {
  color: #80868b;
  margin-bottom: 0px;
}

.service-draft-row {
  /* Rectangle 49 */

  width: 100%;
  height: 72px;

  /* grey/white (#fff) */

  background: #ffffff;
  /* grey/300 ✔ */

  border: 1px solid #dadce0;
  box-sizing: border-box;
  border-radius: 8px;

  border-left-color: #ffa500;
  border-left-width: 4px;

  margin-left: auto;
  margin-right: auto;

  padding-left: 9px;
  padding-right: 9px;
}

.service-draft-row.empty {
  border-left-color: #dadce0;
  border-left-width: 1px;
}

h1.list {
  margin-bottom: 40px;
}

h2.list {
  font-family: Google Sans;
  font-size: 18px;
  line-height: 24px;
  font-weight: normal;
  /* identical to box height, or 133% */

  /* grey/800 ✔ */

  color: #3c4043;
}

table.table td {
  height: 60px;
}

div.search.input-group {
  width: 250px;
}

.search.input-group span.input-group-text {
  background-color: #fff;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  padding-right: 0;
  border-right: none;
}

.search.input-group:focus-within {
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
  border-radius: 0.3em;
}

.search.input-group label {
  display: flex;
  margin: 0;
}

.search.input-group input {
  border-left: none;
  margin: 0;
}

.search.input-group input:focus {
  border-color: #ced4da;
  box-shadow: none;
}
