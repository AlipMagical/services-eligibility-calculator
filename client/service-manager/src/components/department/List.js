/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { defaultLocale, localeLanguage } from '../../Locales.js';

import { AuthContext } from '../../App.js';
import Card from '../../common/components/service/Card.js';
import Create from './Create';
import Show from './Show';
import Update from './Update';
import { base64URLEncode } from '../../common/base64URL';
import { hasRole } from '../../common/utils/index.js';
import { nonDefaultLocales } from '../namedTranslatedEntity.js';
import { useTranslatedDepartments } from '../../hooks';

function ListRow({ department, translations, onManage }) {
  return (
    <div
      key={department['@id']}
      className="d-flex justify-content-between mb-3"
    >
      <div>
        <span>{department['name']}</span>
        {translations && (
          <>
            <span className="ml-1">
              (
              {Object.entries(translations).map(([locale, translation]) => (
                <i key={locale}>
                  {localeLanguage(locale)}: {translation['name']}
                </i>
              ))}
              )
            </span>
          </>
        )}
      </div>{' '}
      <div>
        {onManage && (
          <button className="btn btn-blue-text pull-right" onClick={onManage}>
            Manage
          </button>
        )}
      </div>
    </div>
  );
}

export default function List() {
  const { departments, fetchDepartments } = useTranslatedDepartments();
  const { loggedInUser } = useContext(AuthContext);
  const isAdmin = hasRole(loggedInUser, 'ROLE_ADMIN');
  const [showCreate, setShowCreate] = useState(false);
  const [showEdit, setShowEdit] = useState(false);
  const [departmentId, setDepartmentId] = useState(undefined);

  return (
    <Card
      header={
        <div className="d-flex justify-content-between">
          <div>Departments</div>
          <div>
            {isAdmin && (
              <button
                className="btn btn-primary"
                onClick={() => setShowCreate(true)}
              >
                Add department
              </button>
            )}
          </div>
        </div>
      }
    >
      {showCreate && (
        <Create
          onCancel={() => {
            setShowCreate(false);
            fetchDepartments();
          }}
        />
      )}
      {departmentId && !showEdit && (
        <Show
          onCancel={() => {
            setDepartmentId(undefined);
            fetchDepartments();
          }}
          departmentId={departmentId}
          onEdit={() => setShowEdit(true)}
        />
      )}
      {departmentId && showEdit && (
        <Update
          departmentId={departmentId}
          onCancel={() => {
            setShowEdit(false);
            fetchDepartments();
          }}
          onRemove={() => {
            // Fully unwind the stack of modals so none of them try to
            // display the deleted item.
            setShowEdit(false);
            setDepartmentId(undefined);
          }}
        />
      )}
      {departments &&
        departments[defaultLocale] &&
        departments[defaultLocale].map(department => (
          <ListRow
            key={department['@id']}
            department={department}
            translations={nonDefaultLocales(department['@id'], departments)}
            onManage={
              isAdmin &&
              (() => {
                setDepartmentId(base64URLEncode(department['@id']));
              })
            }
          />
        ))}
    </Card>
  );
}
