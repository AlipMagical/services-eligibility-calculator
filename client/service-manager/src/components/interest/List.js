/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { defaultLocale, localeLanguage } from '../../Locales.js';

import { AuthContext } from '../../App.js';
import Card from '../../common/components/service/Card.js';
import Create from './Create';
import Show from './Show';
import Update from './Update';
import { base64URLEncode } from '../../common/base64URL';
import { hasRole } from '../../common/utils/index.js';
import { nonDefaultLocales } from '../namedTranslatedEntity.js';
import { useTranslatedInterests } from '../../hooks';

function ListRow({ interest, translations, onManage }) {
  return (
    <div key={interest['@id']} className="d-flex justify-content-between mb-3">
      <div>
        <i className="material-icons align-middle">
          {interest['materialIcon']}
        </i>
        <span className="align-middle ml-3">{interest['name']}</span>
        {translations && (
          <>
            <span className="align-middle ml-1">
              (
              {Object.entries(translations).map(([locale, translation]) => (
                <i key={locale}>
                  {localeLanguage(locale)}: {translation['name']}
                </i>
              ))}
              )
            </span>
          </>
        )}
      </div>
      <div>
        {onManage && (
          <button className="btn btn-blue-text" onClick={onManage}>
            Manage
          </button>
        )}
      </div>
    </div>
  );
}

export default function List() {
  const { interests, fetchInterests } = useTranslatedInterests();
  const { loggedInUser } = useContext(AuthContext);
  const isAdmin = hasRole(loggedInUser, 'ROLE_ADMIN');
  const [showCreate, setShowCreate] = useState(false);
  const [showEdit, setShowEdit] = useState(false);
  const [interestId, setInterestId] = useState(undefined);

  return (
    <Card
      header={
        <div className="d-flex justify-content-between mb-0">
          <div>Interests</div>
          <div>
            {isAdmin && (
              <button
                className="btn btn-primary"
                onClick={() => setShowCreate(true)}
              >
                Add interest
              </button>
            )}
          </div>
        </div>
      }
    >
      {showCreate && (
        <Create
          onCancel={() => {
            setShowCreate(false);
            fetchInterests();
          }}
        />
      )}
      {interestId && !showEdit && (
        <Show
          onCancel={() => {
            setInterestId(undefined);
            fetchInterests();
          }}
          interestId={interestId}
          onEdit={() => setShowEdit(true)}
        />
      )}
      {interestId && showEdit && (
        <Update
          interestId={interestId}
          onCancel={() => {
            setShowEdit(false);
            fetchInterests();
          }}
          onRemove={() => {
            // Fully unwind the stack of modals so none of them try to
            // display the deleted item.
            setShowEdit(false);
            setInterestId(undefined);
          }}
        />
      )}
      {interests &&
        interests[defaultLocale] &&
        interests[defaultLocale].map(interest => (
          <ListRow
            key={interest['@id']}
            interest={interest}
            translations={nonDefaultLocales(interest['@id'], interests)}
            onManage={
              isAdmin &&
              (() => {
                setInterestId(base64URLEncode(interest['@id']));
              })
            }
          />
        ))}
    </Card>
  );
}
