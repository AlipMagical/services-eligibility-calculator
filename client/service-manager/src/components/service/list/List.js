/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { useArchivedServices, useServices } from '../../../hooks';

import { AuthContext } from '../../../App';
import DraftsList from './DraftsList';
import { Helmet } from 'react-helmet';
import { Navbar } from 'react-bootstrap';
import SearchBar from './Search';
import ServicesTable from './ServicesTable';

export default function ListServices() {
  const { services, fetchServices } = useServices();
  const { archivedServices, fetchArchivedServices } = useArchivedServices();
  const { isLoggedIn } = useContext(AuthContext);
  const [wasloggedIn, setWasLoggedIn] = useState(isLoggedIn);
  const [lastSearch, setLastSearch] = useState('');

  if (isLoggedIn !== wasloggedIn) {
    setWasLoggedIn(isLoggedIn);
    fetchServices();
    fetchArchivedServices();
  }

  return (
    <>
      <Helmet>
        <title>{process.env.REACT_APP_MANAGER_TITLE}</title>
      </Helmet>
      <Navbar className="service-header">
        <div className="w-100">
          <h1 className="list float-left">
            {process.env.REACT_APP_MANAGER_TITLE}
          </h1>
          <SearchBar onSearch={setLastSearch} />
        </div>
      </Navbar>

      {isLoggedIn && <DraftsList lastSearch={lastSearch} />}
      {services && (
        <ServicesTable services={services} lastSearch={lastSearch} />
      )}
      {archivedServices && (
        <ServicesTable
          services={archivedServices}
          lastSearch={lastSearch}
          archived
        />
      )}
    </>
  );
}
