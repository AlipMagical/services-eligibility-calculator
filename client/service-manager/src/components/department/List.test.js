/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { axe, toHaveNoViolations } from 'jest-axe';

import { AuthContext } from '../../App';
import List from './List';
import { MemoryRouter } from 'react-router';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { mockDepartmentsJson } from '../../../../testing/mockData';
import { render } from 'react-dom';

expect.extend(toHaveNoViolations);

it('renders list departments view', async () => {
  const container = document.createElement('div');

  fetch.resetMocks();
  fetch.doMockOnceIf(
    url => url.includes('/api/departments?locale=en'),
    mockDepartmentsJson
  );
  fetch.doMockOnceIf(
    url => url.includes('/api/departments?locale=es'),
    mockDepartmentsJson
  );

  await act(async () => {
    render(
      <MemoryRouter>
        <AuthContext.Provider
          value={{
            isLoggedIn: true,
            loggedInUser: { username: 'admin', roles: ['ROLE_ADMIN'] }
          }}
        >
          <List />
        </AuthContext.Provider>
      </MemoryRouter>,
      container
    );
  });

  expect(container).toMatchSnapshot();
  expect(await axe(container)).toHaveNoViolations();
});
