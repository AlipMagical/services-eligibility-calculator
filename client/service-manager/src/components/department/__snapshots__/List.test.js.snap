// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list departments view 1`] = `
<div>
  <div
    class="service-card card"
    tabindex="-1"
  >
    <div
      class="card-header"
    >
      <div
        class="d-flex justify-content-between"
      >
        <div>
          Departments
        </div>
        <div>
          <button
            class="btn btn-primary"
          >
            Add department
          </button>
        </div>
      </div>
    </div>
    <div
      class="card-body"
    >
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            City Attorney
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              City Attorney
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            City Clerk's Office
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              City Clerk's Office
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            City Council
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              City Council
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            City Manager's Office
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              City Manager's Office
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Climate Initiatives
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Climate Initiatives
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Communication
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Communication
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Community Vitality
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Community Vitality
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Energy Strategy
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Energy Strategy
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Finance
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Finance
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Fire
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Fire
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Housing and Human Services
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Housing and Human Services
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Human Resources
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Human Resources
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Innovation and Technology
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Innovation and Technology
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Library and Arts
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Library and Arts
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Municipal Court
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Municipal Court
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            OSMP
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              OSMP
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Other
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Other
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Parks and Recreation
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Parks and Recreation
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Planning
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Planning
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Police
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Police
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
      <div
        class="d-flex justify-content-between mb-3"
      >
        <div>
          <span>
            Public Works
          </span>
          <span
            class="ml-1"
          >
            (
            <i>
              Spanish
              : 
              Public Works
            </i>
            )
          </span>
        </div>
         
        <div>
          <button
            class="btn btn-blue-text pull-right"
          >
            Manage
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
